core/libs/commonwealth/commonwealth/settings/manager.py

```diff
@@ -0,0 +1,120 @@
+
+        self.project_name = project_name
+        self.config_path = (
+            config_path.joinpath(project_name)
+            if config_path
+            else pathlib.Path(appdirs.user_config_dir(self.project_name))
+        )
+        self.config_path.mkdir(parents=True, exist_ok=True)
+        self.settings_type = settings_type
+        self._settings = None
+        logger.debug(
+            f"Starting {project_name} settings with {settings_type.__name__}, configuration path: {config_path}"
+        )
+
+    @property
+    def settings(self) -> Any:
+        """Getter point for settings
+
+        Returns:
+            [self.settings_type]: The settings defined in the constructor
+        """
+        if not self._settings:
+            self.load()
+
+        return self._settings
+
+    @settings.setter
+    def settings(self, value: Any) -> None:
+        """Setter point for settings, save settings in any change
+
+        Args:
+            value ([self.settings_type]): The settings defined in the constructor
+        """
+        if not self._settings:
+            self.load()
+
+        self._settings = value
+        self.save()
+
+    def settings_file_path(self) -> pathlib.Path:
+        """Return the settings file for the version specified in the constructor settings
+
+        Returns:
+            pathlib.Path: Path for the settings file
+        """
+        return self.config_path.joinpath(f"{Manager.SETTINGS_NAME_PREFIX}{self.settings_type.VERSION}.json")
+
+    @staticmethod
+    def load_from_file(settings_type: Any, file_path: pathlib.Path) -> Any:
+        """Load settings from a generic location and settings type
+
+        Args:
+            settings_type (BaseSettings): Settings type that inherits from BaseSettings.
+            file_path (pathlib.Path): Path for a valid settings file
+
+        Returns:
+            Any: The settings based on settings_type
+        """
+        assert issubclass(settings_type, settings.BaseSettings), "settings_type should use BaseSettings as subclass"
+
+        settings_data = settings_type()
+
+        if file_path.exists():
+            settings_data.load(file_path)
+        else:
+            settings_data.save(file_path)
+
+        return settings_data
+
+    def save(self) -> None:
+        """Save settings"""
+        self.settings.save(self.settings_file_path())
+
+    def load(self) -> None:
+        """Load settings"""
+
+        def get_settings_version_from_string(string: str) -> int:
+            result = re.search(f"{Manager.SETTINGS_NAME_PREFIX}(\\d+)", string)
+            assert result
+            assert len(result.groups()) == 1
+            return int(result.group(1))
+
+        # Get all possible settings candidates and sort it by version
+        valid_files = [
+            possible_file
+            for possible_file in os.listdir(self.config_path)
```

Comment:
`in self.config_path.iterdir()` would be the modern Pathlib way to do that.

Reply:
good catch!